### PR TITLE
fix(typescript): Allow for using `compilerOptions.moduleResolution`

### PR DESCRIPTION
**Pull request by susnux**
## Rollup Plugin Name: `typescript`

This PR contains:

- [x] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

### Description
* Resolves: #1375 

Align module resolution with the typescript compiler:
If `moduleResolution` is set it should be used and only if it is omitted the module setting might be considered.

### CODE REVIEW

**Review by NotWoods — APPROVED** [review]

**Review by tada5hi — APPROVED** [review]

File: packages/typescript/test/fixtures/nodenext-resolution/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "compilerOptions": {
-    "module": "nodenext"
+    "moduleResolution": "nodenext"
```
Comment (shellscape) [inline packages/typescript/test/fixtures/nodenext-resolution/tsconfig.json:3]: @NotWoods with these tests previously passing using the (incorrect?) `module` property, would this constitute a breaking change?

Comment (susnux) [reply packages/typescript/test/fixtures/nodenext-resolution/tsconfig.json:3]: The tests are still passing, see [nodenext-module/tsconfig.json](https://github.com/rollup/plugins/pull/1453/files/8ad0cd59eff6de147ae086722c5adc56bb1a77d6#diff-18d9a02ea4854ea5ec06af4550b68244da312bc5cb96268c571dfbe215d69c69).
What changed is that `module` is only used if `moduleResolution´ is not set .

File: packages/typescript/src/options/tsconfig.ts [hunk 2]
```diff
@@ -106,11 +106,12 @@ function containsEnumOptions(
 function setModuleResolutionKind(parsedConfig: ParsedCommandLine): ParsedCommandLine {
   const moduleKind = parsedConfig.options.module;
   const moduleResolution =
-    moduleKind === ModuleKind.Node16
+    parsedConfig.options.moduleResolution ||
```
[the remark this answers is not in the document]
Comment (shellscape) [reply packages/typescript/src/options/tsconfig.ts:109]: agreed. @susnux could you please make that change?

[the remark this answers is not in the document]
Comment (susnux) [reply packages/typescript/src/options/tsconfig.ts]: @shellscape done :+1:

**Review by RebeccaStevens — APPROVED** [review]

**Comment by susnux** [comment]
Anything I can help with to get this merged?

**Comment by shellscape** [comment]
Time is super short right now. Thanks for the ping.

**Pull request merged**